java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/db/DBAddConnection.java

```diff
@@ -66,12 +70,66 @@
 @ServiceProvider(service = CodeActionsProvider.class)
 public class DBAddConnection extends CodeActionsProvider {
     public static final String DB_ADD_CONNECTION =  "db.add.connection"; // NOI18N
+    public static final String USER_ID =  "userId"; // NOI18N
+    public static final String PASSWORD =  "password"; // NOI18N
+    public static final String DRIVER =  "driver"; // NOI18N
+    public static final String DB_URL =  "url"; // NOI18N
+    public static final String SCHEMA =  "schema"; // NOI18N
+    public static final String DISPLAY_NAME =  "displayName"; // NOI18N
+    
+    private static final Set<String> COMMANDS = new HashSet<>(Arrays.asList(DB_ADD_CONNECTION));
+    
+    private final Gson gson = new Gson();
 
     @Override
     public CompletableFuture<Object> processCommand(NbCodeLanguageClient client, String command, List<Object> arguments) {
         if (!DB_ADD_CONNECTION.equals(command)) {
             return null;
         }
+        
+        String userId = null;
+        String dbUrl = null;
+        String driverClass = null;
+        
+        final Map m = gson.fromJson(gson.toJson(arguments.get(0)), Map.class);
```

Comment:
This looks fishy I read this as deserialized value -> gson serialization -> gson deserialization. Why? Without further context a java map is the natural mapping of a JS object.

Reply:
This is not resolved, as no answert is provided.

Reply:
This is resolved at line 90: `final Map m = gson.fromJson((JsonObject) arguments.get(0), Map.class);`

Reply:
Seems to be fixed in the last revision